internal/querycoord/task.go

```diff
@@ -1249,6 +1251,12 @@ func (lst *loadSegmentTask) preExecute(context.Context) error {
 		zap.Int64s("segmentIDs", segmentIDs),
 		zap.Int64("loaded nodeID", lst.DstNodeID),
 		zap.Int64("taskID", lst.getTaskID()))
+
+	if err := lst.broker.acquireSegmentsReferLock(ctx, segmentIDs); err != nil {
```

Comment:
I am a bit curious. What about acquiring the segment ref lock when creating the loadSegmentTask? Otherwise, can compaction just happen immediately before preExecute()?

Reply:
`loadSegmentTask ` can be created by loadCollection, loadPartition, handOff , etc. 
If the lock is acquired when the task is created, there will be multiple logic to acquire the lock. So it is better to acquire the lock here. And even if compaction occurs before perExecute, the task will be retried. And querycoord will receive the segment after compaction to cancel the task.

Reply:
BTW the implementation is much tidy than before!
Make the code succinct and tidy is the trick to make it clear and fast

Reply:
If acquire lock failed, querycoord will reschedule this task.